p2p/ping.go

```diff
@@ -0,0 +1,163 @@
+package p2p
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+	ma "github.com/multiformats/go-multiaddr"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	pingInterval   = time.Second
```

Comment:
Would it make sense to have constants like this configurable to be able to tweak them without releasing a new version?

Reply:
Made ping interval configurable. It's a purely troubleshooting feature that I hope will not be needed _too often_, but indeed let's be on the safe side